src/models/Message.js

```diff
@@ -0,0 +1,50 @@
+import User from "./User";
+
+class Message {
+  static mimicRealmMappedPropertiesSchema( msg ) {
+
+    const fromUser = User.mapApiToRealm( msg.from_user );
+    const toUser = User.mapApiToRealm( msg.to_user );
+
+    return {
+      ...msg,
+      subject: msg.subject,
+      body: msg.body,
+      fromUser,
+      toUser,
+      threadId: msg.thread_id
+    };
+  }
+
+  static createMessageForRealm( msg, realm ) {
+    const newMsg = {
+      ...msg,
+        subject: msg.subject,
+        body: msg.body,
+        fromUser: User.mapApiToRealm( msg.from_user, realm ),
+        toUser: User.mapApiToRealm( msg.to_user, realm ),
+        threadId: msg.thread_id
+    };
+
+    // need to append user after the observation realm object has been created
+    //delete newObs.user; //Q: Why do we need to do this?
+    return newMsg;
+  }
+
+
+  //Q: https://api.inaturalist.org/v1/docs/#!/Messages/get_messages shows everything is optional - how do we handle?
+  static schema = {
+    name: "Message",
+    primaryKey: "uuid", //Q: UUID is used instead of ID - is that right?
```

Comment:
I'd go with whatever is in the API documentation here. If we're using `id` in the messages API, best to keep it consistent.

Also, it's fine to have most or all fields be optional in Realm.